pkg/controller/migplan/close.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migplan
+
+import (
+	"context"
+
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+)
+
+// Ensure that resources managed by the MigPlan have been cleaned up
+func (r ReconcileMigPlan) ensureClosed(plan *migapi.MigPlan) error {
+	// Migration Registry
+	err := r.ensureMigRegistriesDelete(plan)
+	if err != nil {
+		return err
+	}
+
+	plan.Status.SetCondition(migapi.Condition{
+		Type:     Closed,
+		Status:   True,
+		Category: Critical,
+		Message:  ClosedMessage,
+	})
+	// Apply changes.
+	plan.Status.EndStagingConditions()
+	err = r.Update(context.TODO(), plan)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Remove the closed condition
+func (r ReconcileMigPlan) deleteClosedCondition(plan *migapi.MigPlan) error {
+	plan.Status.DeleteCondition(Closed)
+
+	// Apply changes.
+	plan.Status.EndStagingConditions()
```

Comment:
Directly deleting the condition instead of staging.  Don't need/want:  `EndStagingConditions()` here.

Reply:
OK. I'll remove it. I was following the closest example I had, which followed this pattern:
https://github.com/fusor/mig-controller/blob/master/pkg/controller/migplan/storage.go#L69
Perhaps this code shouldn't be calling this either.

Reply:
No, it shouldn't.